webpack.config.js

```diff
@@ -0,0 +1,102 @@
+const path = require("path");
+const { CleanWebpackPlugin } = require("clean-webpack-plugin");
+const HTMLWebpuckPlugin = require("html-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+
+const isProd = process.env.NODE_ENV === "production";
+const isDev = !isProd;
+
+const filename = (ext) => {
+  return isDev ? `bundle.${ext}` : `bundle.[hash].${ext}`;
+};
+
+const jsLoaders = () => {
+  const loaders = [
+    {
+      loader: "babel-loader",
+      options: {
+        presets: ["@babel/preset-env"],
+        plugins: ["@babel/plugin-proposal-class-properties"],
+      },
+    },
+  ];
+
+  return loaders;
+};
+
+module.exports = {
+  context: path.resolve(__dirname, "src"),
+  mode: "development",
+  entry: ["@babel/polyfill", "./index.js"],
+  output: {
+    filename: filename("js"),
+    path: path.resolve(__dirname, "dist"),
+  },
+  resolve: {
+    extensions: [".js"],
+    alias: {
+      "@": path.resolve(__dirname, "src"),
+      "@core ": path.resolve(__dirname, "src/core"),
+    },
+  },
+  devtool: isDev ? "source-map" : false,
+  devServer: {
+    port: 3000,
+    hot: isDev,
+  },
+  plugins: [
+    new CleanWebpackPlugin(),
+    new HTMLWebpuckPlugin({
+      template: "index.html",
+      minify: {
+        removeComments: isProd,
+        collapseWhitespace: isProd,
+      },
+    }),
+    new CopyPlugin({
+      patterns: [
+        {
+          from: path.resolve(__dirname, "src/favicon.ico"),
+          to: path.resolve(__dirname, "dist"),
+        },
+      ],
+    }),
+    new MiniCssExtractPlugin({
+      filename: filename("css"),
+    }),
+  ],
+  module: {
+    rules: [
+      {
+        test: /\.s[ac]ss$/i,
+        use: [
+          {
+            loader: MiniCssExtractPlugin.loader,
+            options: {
+              hmr: isDev,
+              reloadAll: true,
+            },
+          },
+          "css-loader",
+          "sass-loader",
+        ],
+      },
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: jsLoaders(),
```

Comment:
i`ll recommend to check jsx files too. We will write our react code in jsx, if you won't choose typescript